pkg/controller/cluster/cluster_reconcile_test.go

```diff
@@ -363,3 +363,73 @@ func TestReconcileCluster_getServiceAccountToken(t *testing.T) {
 		})
 	}
 }
+
+func TestClusterCatalogTemplate(t *testing.T) {
```

Comment:
Missing tests for error cases.

Is functional testing the only thing we can do here? I would prefer to also have some unit tests.